File: server/model/delicacy.js
const Sequelize = require('sequelize')
const MySequelize = require('../config/mysqlConfig')

const delicacyModel = MySequelize.define('delicacy', {
  id: {
    type: Sequelize.STRING(19),
    field:  'id',
    primaryKey: true
  },
  name: {
    type: Sequelize.STRING(20),
    field:  'name',
    allowNull: false
  },
  description: {
    type: Sequelize.STRING(500),
    field:  'description',
    allowNull: false
  },
  cover: {
    type: Sequelize.STRING(255),
    field:  'cover',
    allowNull: false
  },
  recommend_rank: {
    type: Sequelize.TINYINT,
    field:  'recommend_rank',
    allowNull: false
  },
  is_deleted: {
    type: Sequelize.TINYINT,
    field:  'is_deleted',
    allowNull: false
  },
  create_time: {
    type: Sequelize.DATE,
    field:  'create_time',
    allowNull: false
  },
  modify_time: {
    type: Sequelize.DATE,
    field:  'modify_time',
    allowNull: false
  },
})

module.exports = delicacyModel

File: src/main/java/com/carolin_violet/travel_system/bean/Delicacy.java
package com.carolin_violet.travel_system.bean;

import com.baomidou.mybatisplus.annotation.*;

import java.util.Date;
import java.io.Serializable;

import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.experimental.Accessors;

/**
 * <p>
 * 
 * </p>
 *
 * @author carolin-violet
 * @since 2022-05-22
 */
@Data
@EqualsAndHashCode(callSuper = false)
@Accessors(chain = true)
public class Delicacy implements Serializable {

    private static final long serialVersionUID = 1L;

    /**
     * 美食id
     */
    @TableId(value = "id", type = IdType.ID_WORKER_STR)
    private String id;

    /**
     * 美食名称
     */
    private String name;

    /**
     * 美食描述
     */
    private String description;

    /**
     * 美食图片
     */
    private String cover;

    /**
     * 推荐指数(1,2,3,4,5)
     */
    private Integer recommendRank;

    /**
     * 逻辑删除
     */
    @TableLogic
    @TableField(fill = FieldFill.INSERT)
    private Integer isDeleted;

    /**
     * 创建时间
     */
    @TableField(fill = FieldFill.INSERT)
    private Date createTime;

    /**
     * 修改时间
     */
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private Date modifyTime;


}
